scrapers_library/data_portals/muckrock/muckrock_tools/README.md

```diff
@@ -0,0 +1,88 @@
+# Muckrock Toolkit
+
+## Description
+
+This repo provides tools for searching Muckrock FOIA requests, it includes scripts for downloading data from MuckRock, generating CSV files per PDAP database requirements, and automatic labeling
+
+## Installation
+
+### 1. Clone the `scrapers` repository and navigate to the `muckrock_tools` directory.
+
+```
+git clone git@github.com:Police-Data-Accessibility-Project/scrapers.git
+cd scrapers/scrapers_library/data_portals/muckrock/muckrock_tools
+```
+
+### 2. Create a virtual environment.
+
+If you don't already have virtualenv, install the package:
+
+```
+
+pip install virtualenv
+
+```
+
+Then run the following command to create a virtual environment (ensure the python version is as below):
+
+```
+
+virtualenv -p python3.12 venv
+
+```
+
+### 3. Activate the virtual environment.
+
+```
+
+source venv/bin/activate
+
+```
+
+### 4. Install dependencies.
+
+```
+
+pip install -r requirements.txt
+
+```
+
+## Uses
+
+### 1. Simple Search Term
+
+- `muck_get.py`
+- script to perform searches on MuckRock's database, by matching a search string to title of request. Search is slow due to rate limiting (cannot multi thread around it).
+
+### 2. Clone Muckrock database & search locally
+
+~~- `download_muckrock_foia.py` `search_local_foia_json.py`~~ (deprecated)
+
+- scripts to clone the MuckRock foia requests collection for fast local querying (total size <2GB at present)
+
+- `create_foia_data_db.py` creates and populates a SQLite database (`foia_data.db`) with all MuckRock foia requests. Various errors outside the scope of this script may occur; a counter (`last_page_fetched.txt`) is created to keep track of the most recent page fetched and inserted into the database. If the program exits prematurely, simply run `create_foia_data_db.py` again to continue where you left off. A log file is created to capture errors for later reference.
+
+- After `foia_data.db` is created, run `search_foia_data_db.py`, which receives a search string as input and outputs a JSON file with all related foia requests for later processing by `generate_detailed_muckrock_csv.py`. For example,
+
+```
+python3 search_foia_data_db.py --search_for "use of force"
```

Comment:
Now I tried running this command, and got the following error:

```
py search_foia_data_db.py --search_for "use of force"
'Searching foia_data.db for 'use of force'...
Traceback (most recent call last):
  File "C:\Users\maxac\OneDrive\CODING\scrapers\.venv\Lib\site-packages\pandas\io\sql.py", line 2674, in execute
    cur.execute(sql, *args)
sqlite3.OperationalError: no such table: results

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\maxac\OneDrive\CODING\scrapers\scrapers_library\data_portals\muckrock\muckrock_tools\search_foia_data_db.py", line 69, in <module>
    main()
  File "C:\Users\maxac\OneDrive\CODING\scrapers\scrapers_library\data_portals\muckrock\muckrock_tools\search_foia_data_db.py", line 55, in main
    df = search_foia_db(search_string)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maxac\OneDrive\CODING\scrapers\scrapers_library\data_portals\muckrock\muckrock_tools\search_foia_data_db.py", line 29, in search_foia_db
    df = pd.read_sql_query(query, conn, params=params)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maxac\OneDrive\CODING\scrapers\.venv\Lib\site-packages\pandas\io\sql.py", line 526, in read_sql_query
    return pandas_sql.read_query(
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maxac\OneDrive\CODING\scrapers\.venv\Lib\site-packages\pandas\io\sql.py", line 2738, in read_query
    cursor = self.execute(sql, params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\maxac\OneDrive\CODING\scrapers\.venv\Lib\site-packages\pandas\io\sql.py", line 2686, in execute
    raise ex from exc
pandas.errors.DatabaseError: Execution failed on sql '
                SELECT * FROM results
                WHERE (title LIKE ? OR tags LIKE ?)
                AND (status = 'done')
                ': no such table: results
```

I'm _guessing_ that the table in question is created after this part of the script is executed, or else it was created off-script?